java/org/apache/catalina/valves/AbstractAccessLogValve.java

```diff
@@ -1506,6 +1508,38 @@ public void addElement(CharArrayWriter buf, Date date, Request request,
         }
     }
 
+    /**
+     * Write connection status when response is completed - %X
+     */
+    protected static class ConnectionStatusElement implements AccessLogElement {
+        @Override
+        public void addElement(CharArrayWriter buf, Date date, Request request, Response response, long time) {
+            if (response != null && request != null) {
+                // Check for connection aborted cond
+                boolean isConnAborted = false;
+                if (response.isError()) {
+                    Throwable ex = (Throwable)request.getAttribute(RequestDispatcher.ERROR_EXCEPTION);
+                    if (ex instanceof ClientAbortException) {
+                        isConnAborted = true;
+                        buf.append('X');
+                    }
+                }
+
+                // Check whether connection is keep-alive or not
+                if (!isConnAborted) {
+                    if (org.apache.coyote.http11.Constants.KEEPALIVE.equals(
+                            request.getHeader(org.apache.coyote.http11.Constants.CONNECTION))) {
```

Comment:
The log message is intended to show if the connection remains in keep-alive after the current request finishes. You want to look at the Connection header (for close) on the response here to mimic the behaviour described for httpd.